parsers/SG.py

```diff
@@ -197,22 +206,24 @@ def fetch_production(
             logger.warning(msg)
             generation_by_type["unknown"] += gen_mw
 
-    generation_by_type["solar"] = get_solar(requests_obj, logger)
+    generation_by_type["solar"] = 0#get_solar(requests_obj, logger)
 
     # some generation methods that are not used in Singapore
     generation_by_type.update({"nuclear": 0, "wind": 0, "hydro": 0})
-
-    return {
-        "datetime": sg_data_to_datetime(data),
-        "zoneKey": zone_key,
-        "production": generation_by_type,
-        "storage": {},  # there is no known electricity storage in Singapore
-        "source": "emcsg.com, ema.gov.sg",
-    }
+    mix = ProductionMix(**generation_by_type)
```

Comment:
While functional the purpose of the Parser classes is to unify and remove unneeded code from the parsers. 

For example this piece of code on line 197:
```python
        if mapped_type:
            generation_by_type[TYPE_MAPPINGS[gen_type]] += gen_mw
```

Should be:
```python
        if mapped_type:
            mix.add_value([TYPE_MAPPINGS[gen_type]], gen_mw)
```
Where mix is the production mix.
This is to ensure we reduce the floating point errors we might introduce from doing math operations and to ensure that if one of the values are negative it's corrected before it's summed with the other values.